setup.go

```diff
@@ -0,0 +1,132 @@
+package influxdb
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"net/http"
+)
+
+// Setup sets up a new influxdb server.
+// It requires a client be set up with a username and password.
+// If successful will add a token to the client.
+// RetentionPeriodHrs of zero will result in infinite retention.
+func (c *Client) Setup(ctx context.Context, bucket, org string, RetentionPeriodHrs int) (*SetupResult, error) {
```

Comment:
normally a time.Duration would make sense, but I switched too an int, because the API expects an hours int and I do not want to imply that the period would be more fine grained than hours.